pkg/util/util.go

```diff
@@ -25,11 +26,47 @@ func GetCertPool(paths []string) (*x509.CertPool, error) {
 	return pool, nil
 }
 
+// GetRequestProto return the request host header or X-Forwarded-Proto if present
+func GetRequestProto(req *http.Request) string {
+	proto := req.Header.Get("X-Forwarded-Proto")
+	if proto == "" {
+		proto = req.URL.Scheme
+	}
+	return strings.ToLower(proto)
+}
+
 // GetRequestHost return the request host header or X-Forwarded-Host if present
 func GetRequestHost(req *http.Request) string {
 	host := req.Header.Get("X-Forwarded-Host")
 	if host == "" {
 		host = req.Host
 	}
-	return host
+	return strings.ToLower(host)
```

Comment:
I don't think we should tamper with these -- at least not at this layer.  If a consumer of these needs them lower cased, I'd prefer they do that. This method is used in a few areas of the code.

Reply:
Alright! I'll revert back in a moment.